File: extrinsics/tickets.go
package extrinsics

import (
	"github.com/ascrivener/jam/types"
)

type Tickets []Ticket

type Ticket struct {
	TicketEntryIndex    types.TicketEntryIndex
	TicketValidityProof types.BandersnatchRingVRFProof
}

File: src/block/extrinsics/tickets.rs
// src/extrinsics/tickets.rs

use crate::types::{BandersnatchRingVRFProof, TicketEntryIndex};

#[derive(Debug, Clone)]
pub struct Ticket {
    pub entry_index: TicketEntryIndex,
    pub validity_proof: BandersnatchRingVRFProof,
}

pub type Tickets = Vec<Ticket>;
